src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt

```diff
@@ -78,27 +87,45 @@ fun zipDirectory(directory: File): ByteArray {
     return byteOutput.toByteArray()
 }
 
-class ResultReceiver() : CompilerFleetMessageReceiver() {
+fun getTaskId(data: ByteArray): String {
+    val messageDigest = MessageDigest.getInstance("SHA-1")
+
+    return String(messageDigest.digest(data))
+}
+
+private val LOGGER = LoggerFactory.getLogger("ResultReceiver")
+
+class ResultReceiver(
+        private val outputFile: File,
+        private val expectedTaskId: String
+) : CompilerFleetMessageReceiver() {
     override fun processMessage(message: PubsubMessage) {
         val result = CompilerFleet.CompilerFleetResult.parseFrom(message.data)
 
-        println(result.taskId)
-        println(String(result.resultBytes.toByteArray()))
+        if (result.taskId != expectedTaskId) {
+            LOGGER.error("Task ids don't match: \nexpected:{}, \nactual:{}", expectedTaskId, result.taskId)
+            return
+        }
+
+        FileUtils.writeByteArrayToFile(outputFile, result.resultBytes.toByteArray())
     }
 }
 
 fun main(args: Array<String>) {
     val parsedArgs = ArgParser(args).parseInto(::PublisherArgs)
-    val directory = parsedArgs.directory
-    val zippedData = zipDirectory(File(directory))
+    val directory = Paths.get(parsedArgs.directory)
+    val zippedData = zipDirectory(directory.toFile())
     val topic = parsedArgs.publishTopic
+    val rootFileName = parsedArgs.rootFileName
+    val rootFile = directory.resolve(rootFileName).toFile()
```

Comment:
This complex  construction is only used to get the file name which is already available one line above, no?

Reply:
Yes, indeed. Done